### DISCUSSION TITLE
Way to throw an error from within SQL

### DISCUSSION THREAD

**Discussion by zenazn** [duckdb/duckdb#9038] [Ideas]
Similar to the motivation in #3804, I have some messy data, and I'd like to parse it. I might use `try_cast(...)` or `try_strptime(...)`, or a `CASE` statement with some regexes. However, if the data does *not* parse, I'd like to throw an error / stop the query.

One workaround is to run a known-invalid cast, like `CAST('error: your data sucks' AS BOOL)`. This seems to work even in e.g., `COALESCE` contexts, which is nice:
<img width="784" alt="Screenshot 2023-09-21 at 3 31 29 PM" src="https://github.com/duckdb/duckdb/assets/12960/34caadfd-32e4-4157-bd77-a29f58ab9767">

But this feels like a hack! I'd ideally love to write e.g., `coalesce(try_strptime('2022-01-01', '%Y-%m'), error('your data sucks'))`

**Comment by Mause** [comment]
The error function already exists actually, but you may have to use a case statement instead

**Comment by zenazn** [comment]
oh! that's perfect. I'll file an issue about documenting the function, but this works great for my use case. thank you!
